src/main.rs

```diff
@@ -210,6 +213,26 @@ impl BootloaderConnection {
         Ok(())
     }
 
+    fn check_crc(&self, data: &[u8], received_crc: u32, initial: u32) -> Result<u32> {
+
+        let mut digest = crc32::Digest::new_with_initial(crc32::IEEE, initial);
+        digest.write(data);
+        let expected_crc = digest.sum32() & 0xffffffff;
+
+        match expected_crc == received_crc {
```

Comment:
wouldn't `if` be simpler here?

Reply:
I got a little `match`-excited and wasn't sure what the most idiomatic choice would be :D